Alex/intake

AlexCook75921312312312 opened the pull request:
I think that because of clean up the code is now incompatible with the drivetrain stuff

AlexCook75921312312312 committed:
Maps 4 motors

Maps 4 of the motors related to the intake and launching subsystem
Sets all of the ports to zero for now

AlexCook75921312312312 committed:
Creates a new command to alter flywheel speed

Creates a command to set the flywheels for intake and launching
to a speed that would allow them to intake the ball.

AlexCook75921312312312 committed:
Creates a command to stop the intake wheels

This command stops the intake flywheels from spinning
in case that was ever necessary

AlexCook75921312312312 committed:
Creates the intake/launcher subsystem

This subsystem has additional setup for the motors
and defines some method to control the motors and the
intake elevation

AlexCook75921312312312 committed:
Removes (mostly) useless comments

AlexCook75921312312312 committed:
Names the command correctly

AlexCook75921312312312 committed:
Adds some new robot components and ports

Adds a limit switch, a solenoid, and a compressor to RobotMap

AlexCook75921312312312 committed:
Adds new methods for pneumatics

Adds a method to extend the cylinder and retract the cylinder
Adds a limit switch to determine when the ball has been
completely sucked in.

AlexCook75921312312312 committed:
Changes end condition on command

Changes the end condition to end once the limit switch is 
pressed. In the future there might need to be an emergency 
button the driver can press in case the limit switch doesn't 
press

AlexCook75921312312312 committed:
Adds a pneumatic command

Adds a command to retract the punching arm, as
to not interfere with the picking up of future balls

AlexCook75921312312312 committed:
Adds a pneumatic command

Adds a command to extend the pneumatic cylinder 
and punch the ball into the launch flywheels

AlexCook75921312312312 committed:
Adds intake launcher subsystem and related commands
Adds joystick and joystick buttons in OI
Still needs joystick control for the launcher

phroa commented on src/org/usfirst/frc/team4915/stronghold/commands/ExtendLauncherCylinderCommand.java:
```diff
@@ -0,0 +1,32 @@
+        requires(Robot.intakeLauncher);
+    }
+
+    protected void initialize() {
```
consider running `Code -> Clean Up...` on your src folder to automatically add the `@Override` annotations that are missing from these methods.

phroa commented on src/org/usfirst/frc/team4915/stronghold/subsystems/DriveTrain.java:
```diff
@@ -36,8 +34,8 @@ public double modifyThrottle() {
 
     private void setMaxOutput(double topSpeed) {
         // TODO Auto-generated method stub
```
I think you left a `TODO` here

AlexCook75921312312312 committed:
Adds a button that will toggle auto-refill for the launcher
compressor. Includes a new command and a new method in
IntakeLauncher

AlexCook75921312312312 committed:
Adds some code that allows the elevator to be controled with a joystick
Elevator will only move with the joystick while the button is held

AlexCook75921312312312 committed:
Adds necessary override statements

AlexCook75921312312312 commented:
When I used clean code to put the override statements back it seems to have had the side effect of filling my code with "this.". I will fix that during the next pull

phroa commented:
that's actually intentional

phroa commented:
oh, it looks like you committed the `FRCUserProgram.jar` in the `dist` folder. that should probably go.

AlexCook75921312312312 commented:
I have no memory of changing that

AlexCook75921312312312 committed:
Sets unkown ports to -1 to avoid confusion.
Adds support for an encoder to make launcher motion more accurate
Adds more comments
Changes some packaging stuff
Adds a new command for waiting in order to give flywheels spinup time
Changes the way SetElevatorHeightWorks
Adds a new command to calibrate the newly aquired encoder, though that
command is not used so far

AlexCook75921312312312 committed:
Removes a line about a scaling arm that doesn't exist

AlexCook75921312312312 committed:
Puts the wait command into the correct package

AlexCook75921312312312 committed:
Adds some TODO comments, and removes some scaling arm stuff that shouldn't
be there

AlexCook75921312312312 committed:
Tweaks various things about the launcher system
Changes the launcherMotors to a single motor
Changes the name of launcherMotor to aimMotor
Adds some TODO comments to unknown ports
Removed useless compresser
Removed things related to an encoder

AlexCook75921312312312 committed:
Removes some now irrelevant comments

AlexCook75921312312312 committed:
Makes commands not interrupt eachother
Removes more useless stuff

AlexCook75921312312312 committed:
Adds more formatting stuff
